buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoringOld.groovy

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.gradle
+
+import org.gradle.api.Project
+
+/**
+ * Utilities for working with our vendored version of gRPC.
+ */
+class GrpcVendoringOld {
```

Comment:
Why introducing this “-Old” class? Is the reason worth adding in the Javadoc?

Reply:
This is because we have to upgrade the gRPC version in two steps: 
1) Upgrade the gRPC vendor to 1.26.0. The Beam classes still depend on gRPC vendor 1.21.0 in this step and so we have to introduce GrpcVendoringOld to make this possible. This is also the way taked  previously: https://github.com/apache/beam/commit/137953b0d15d2942b051a69621fecd53a1bda141.
2) Once the gRPC vendor is released, then we need to migrade the Beam classes to use 1.26.0. In this step, we can remove the GrpcVendoringOld introduced in step 1)

Reply:
vendored gRPC 1.21.0 is already released and is not part of the normal build process anymore so I agree with @suztomo that it should be unnecessary to add `GrpcVendoringOld`.

Reply:
(My question was a pure question. Not an objection) Good to know the vendored artifact has different release cycle. I found [`org.apache.beam:beam-vendor-grpc-1_21_0:0.1`](https://search.maven.org/artifact/org.apache.beam/beam-vendor-grpc-1_21_0/0.1/jar) in Maven Central.

@lukecwik Is that special release process documented? (From 137953b, I assume you handled the release process last in June this year) If not, would you add one in https://beam.apache.org/contribute/release-guide/ ?

@sunjincheng121 Thanks for the explanation. For future reference, would you add source code comment that the vendored gRPC has separate release process?

Reply:
@suztomo I would like to do the improvement but I do not found a better place to add the comment(The separate release doesn't apply for GRPC, but also other vendors, such as calcite, guava, etc.). Could you please give me more detail thoughts? How about making a bit improvement in https://beam.apache.org/contribute/release-guide/ for separate release process?

Reply:
> How about making a bit improvement in https://beam.apache.org/contribute/release-guide/ for separate release process?

Yes, I expect that’s the first step (Can Lukasz take this?). Then add the URL to “vendor/grpc-1_26_0/build.gradle”.

Reply:
There is a release guide for vendored artifacts here: https://s.apache.org/beam-release-vendored-artifacts (would greatly appreciate it if someone could add it to the beam site since it is the remaining work on https://issues.apache.org/jira/browse/BEAM-6438).

Also, the linkage tool is useful for all dependency upgrades which is useful for more then just when creating vendored dependencies so documentation should go in some shared location.

Reply:
You could add some helpful links to `vendor/README.md`.

Also, now I remember/understand why `GrpcVendoringOld` needs to exist (since the proto generated code needs to be updated to use the vendored version)

Reply:
Instead of new/old, use version suffixes `GrpcVendoring_1_21_0` and `GrpcVendoring_1_26_0`

Reply:
+1 to use `GrpcVendoring_1_21_0` and `GrpcVendoring_1_26_0`. Have also added `vendor/README.md` about how to upgrade and validate the vendored dependencies.